metricflow-semantics/metricflow_semantics/query/query_parser.py

```diff
@@ -92,12 +93,13 @@ def __init__(  # noqa: D107
     def parse_and_validate_saved_query(
         self,
         saved_query_parameter: SavedQueryParameter,
-        where_filters: Optional[Sequence[WhereFilter]],
-        limit: Optional[int],
-        time_constraint_start: Optional[datetime.datetime],
-        time_constraint_end: Optional[datetime.datetime],
-        order_by_names: Optional[Sequence[str]],
-        order_by_parameters: Optional[Sequence[OrderByQueryParameter]],
+        where_filters: Optional[Sequence[WhereFilter]] = None,
+        limit: Optional[int] = None,
+        time_constraint_start: Optional[datetime.datetime] = None,
+        time_constraint_end: Optional[datetime.datetime] = None,
+        order_by_names: Optional[Sequence[str]] = None,
+        order_by_parameters: Optional[Sequence[OrderByQueryParameter]] = None,
+        dedupe: bool = True,
```

Comment:
wonder if this is hard to infer what it does based on the name 🤔 would like `dedupe_with_group_by` be better?

Reply:
Ugh naming is hard. I want to call it something like `apply_group_by` (which would be true for metric queries too). But in a lot of places `group_by` is the name of a parameter already so it feels weird, since if `apply_group_by` is false then the `group_by` param name no longer makes sense. But I guess the `group_by` param name is the issue in that case so I shouldn't let that affect how I name this param...